runtime/parachains/src/disputes.rs

```diff
@@ -632,7 +622,7 @@ impl<T: Config> Pallet<T> {
 
 				// This is larger, and will be extracted to the `shared` module for more proper pruning.
 				// TODO: https://github.com/paritytech/polkadot/issues/3469
-				<Included<T>>::remove_prefix(to_prune, None);
+				shared::Pallet::<T>::prune_included_candidates(to_prune);
```

Comment:
The `last_pruned` update logic needs to be changed because pruning here may only be partial.

I suggest having the `shared` module own all the pruning logic anyway. It's weird to have `shared` pruning logic triggered by the disputes module.